.github/workflows/cpp-post-commit-wrapper.yaml

```diff
@@ -13,5 +13,16 @@ jobs:
     secrets: inherit
   cpp-unit-tests:
     needs: build-artifact
-    uses: ./.github/workflows/cpp-post-commit.yaml
     secrets: inherit
+    strategy:
+      fail-fast: false
+      matrix:
+        test-group: [
+          { arch: grayskull, runner-label: E150 },
+          { arch: wormhole_b0, runner-label: N150 },
+          { arch: wormhole_b0, runner-label: N300 },
+        ]
+    uses: ./.github/workflows/cpp-post-commit.yaml
+    with:
+      arch: ${{ matrix.test-group.arch}}
```

Comment:
Could we get some yamllint in here

Reply:
https://github.com/tenstorrent/tt-metal/issues/11034
Would it make sense to talk about it in this issue and fix our yaml formatting issues in future PRs

I've added yamllint and yamlfmt to pre-commit already and gave it a whirl. It works though i'm not sure the default outcome is what we want.

Reply:
@tt-rkim this should be a future PR, file an issue